R/estimateDiversity.R

```diff
@@ -178,13 +188,120 @@ setMethod("estimateDiversity", signature = c(x = "SummarizedExperiment"),
                                          .get_diversity_values,
                                          x = x,
                                          mat = assay(x, abund_values),
+                                         tree = tree,
                                          BPPARAM = BPPARAM,
                                          ...)
         .add_values_to_colData(x, dvrsts, name)
     }
 )
 
+#' @rdname estimateDiversity
+#' 
+#' @param tree A phylogenetic tree that is used to calculate 'faith' index.
+#' By default, it is \code{rowTree(x)}.
+#' 
+#' @export
+setMethod("estimateDiversity", signature = c(x="TreeSummarizedExperiment"),
+    function(x, abund_values = "counts",
+             index = c("shannon","gini_simpson","inverse_simpson",
+                       "coverage", "fisher", "faith"),
+             name = index, tree = rowTree(x), ..., BPPARAM = SerialParam()){
```

Comment:
Since you supply a potential tree with `x`, I think you can drop the tree argument here. Just add:

```r
tree <- rowTree(x)
```

directly in the function. Otherwise the user could overwrite the tree and that doesn't make sense

Reply:
Removed